docs/authorization-configuration.md

```diff
@@ -120,4 +120,26 @@ When Luigi fails to renew the token and then logs out the user, it adds the foll
 - `enabled` means third party cookies are supported by the browser.
 - `not_checked` means that the script was not provided in **thirdPartyCookiesScriptLocation** or it could not be loaded.
 
-The application developer can read these parameters and set a logout page based on them.
\ No newline at end of file
+The application developer can read these parameters and set a logout page based on them.
+
+### Implement a Custom Authentication Provider
+
+You can write your own authentication provider that meets your requirements. 
+
+[oAuth2ImplicitGrant.js](src/providers/oAuth2ImplicitGrant.js) is a good starting point if you don't use an external authentication library.
+After authorization is successful on the auth provider's side it redirects back to `Luigi callback.html` **redirect_uri**. The auth provider verifies the authentication data, saves it in  **localStorage** for Luigi, and redirects to the Luigi main page. 
+
+[openIdConnect.js](src/providers/openIdConnect.js) lazy loads the official `oidc-client` library and is a good starting point if you also depend on external authentication libraries.
+
+Make sure to set the following data in your Auth Provider implementation after authentication.
```

Comment:
```suggestion
Make sure to set the following data in your Authentication Provider implementation, so that it is used after successful authentication.
```
There was something missing in this one - but verify it, please :) And correct if I understand it wrongly.